src/init.ts

```diff
@@ -4,146 +4,235 @@ import fs from 'fs';
-      default: false,
-    },
-  ]);
-
-  const command = mode === supportedModes[0] ? 'npx' : 'docker';
-  const args =
-    mode === supportedModes[0]
-      ? ['-y', '@tigrisdata/tigris-mcp-server', 'run'] // npx command
-      : ['run']; // docker command
-
-  let env: Record<string, string> = {};
-
-  if (useAwsProfiles) {
-    const { awsProfile } = await inquirer.prompt([
+  if (!selectedApplication) {
+    const { application } = await inquirer.prompt([
       {
-        type: 'input',
-        name: 'awsProfile',
-        message: 'Enter AWS profile name:',
-        default: 'default',
+        type: 'list',
+        name: 'application',
+        message: 'Select Application:',
+        choices: Object.keys(applicationConfigMap),
       },
     ]);
-    env.USE_AWS_PROFILES = 'true';
-    env.AWS_PROFILE = awsProfile;
-  } else {
-    const { awsAccessKeyId } = await inquirer.prompt([
+    selectedApplication = application;
+  }
+  let transport = 'remote';
+  let config: MCP_SERVER_STDIO_CONFIG | MCP_SERVER_REMOTE_CONFIG = {};
+
+  if (askForTransport) {
+    const { selectedTransport } = await inquirer.prompt([
       {
-        type: 'input',
-        name: 'awsAccessKeyId',
-        message: 'Enter you AWS/Tigris Access Key Id:',
+        type: 'list',
+        name: 'selectedTransport',
+        message: 'Select Transport:',
+        choices: supportedTransports,
       },
     ]);
-    const { awsSecretAccessKey } = await inquirer.prompt([
+
+    transport = selectedTransport;
+  }
+
+  if (transport === 'stdio') {
+    const { mode } = await inquirer.prompt([
       {
-        type: 'input',
-        name: 'awsSecretAccessKey',
-        message: 'Enter AWS/Tigris Secret Access Key:',
+        type: 'list',
+        name: 'mode',
+        message: 'Run via:',
+        choices: supportedModes,
       },
     ]);
-    env.AWS_ACCESS_KEY_ID = awsAccessKeyId;
-    env.AWS_SECRET_ACCESS_KEY = awsSecretAccessKey;
-  }
 
-  const { awsEndpointUrl } = await inquirer.prompt([
-    {
-      type: 'input',
-      name: 'awsEndpointUrl',
-      message: 'Enter AWS/Tigris S3 Endpoint:',
-      default: 'https://fly.storage.tigris.dev',
-    },
-  ]);
-  env.AWS_ENDPOINT_URL_S3 = awsEndpointUrl;
-
-  if (command === 'docker') {
-    console.log(
-      'Pulling the Tigris MCP Server Docker image. This may take a few minutes...\n',
-    );
-    const downloadImage = spawnSync(
-      'docker',
-      ['image', 'pull', tigrisMcpServerImage],
-      { encoding: 'utf-8' },
-    );
+    const { useAwsProfiles } = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'useAwsProfiles',
+        message: 'Do you want to use AWS profiles?',
+        default: false,
+      },
+    ]);
+
+    const command = mode === supportedModes[0] ? 'npx' : 'docker';
+    const args =
+      mode === supportedModes[0]
+        ? ['-y', '@tigrisdata/tigris-mcp-server', 'run'] // npx command
+        : ['run']; // docker command
```

Comment:
This feels overly golfed. Try something less clever:

```suggestion
    let command: string | null = null;
    let args: string[] | null = null;
    
    if (mode === 'npx') {
        command = 'npx';
        args = ['-y', '@tigrisdata/tigris-mcp-server', 'run'];
    } else if (mode === 'npx') {
        command = 'docker';
        args = ['run'];
    } else {
        throw new Error(`Unsupported execution mode ${mode}, wanted one of ${supportedModes}`);
    }
```

This is longer, but it is much more explicit. Setup code needs to be less clever because it's something users trust implicitly.